spring-kafka/src/test/java/org/springframework/kafka/listener/ShareKafkaMessageListenerContainerUnitTests.java

```diff
@@ -0,0 +1,402 @@
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class ShareKafkaMessageListenerContainerUnitTests {
+
+	@Mock
+	private ShareConsumerFactory<String, String> shareConsumerFactory;
+
+	@Mock
+	private ShareConsumer<String, String> shareConsumer;
+
+	@Mock
+	private MessageListener<String, String> messageListener;
+
+	@Mock
+	private AcknowledgingShareConsumerAwareMessageListener<String, String> ackListener;
+
+	private ContainerProperties containerProperties;
+
+	private ConsumerRecord<String, String> testRecord;
+
+	private ConsumerRecords<String, String> testRecords;
+
+	@Test
+	void shouldConfigureExplicitModeCorrectly() {
+
+		ContainerProperties containerProperties = new ContainerProperties("test-topic");
+		containerProperties.setShareAcknowledgmentMode(ContainerProperties.ShareAcknowledgmentMode.EXPLICIT);
+		containerProperties.setMessageListener(ackListener);
+
+		ShareKafkaMessageListenerContainer<String, String> container =
+				new ShareKafkaMessageListenerContainer<>(shareConsumerFactory, containerProperties);
+
+		assertThat(container.getContainerProperties().getShareAcknowledgmentMode())
+				.isEqualTo(ContainerProperties.ShareAcknowledgmentMode.EXPLICIT);
+	}
+
+	@Test
+	void shouldConfigureImplicitModeByDefault() {
+		ContainerProperties containerProperties = new ContainerProperties("test-topic");
+		containerProperties.setMessageListener(messageListener);
+
+		ShareKafkaMessageListenerContainer<String, String> container =
+				new ShareKafkaMessageListenerContainer<>(shareConsumerFactory, containerProperties);
+
+		assertThat(container.getContainerProperties().getShareAcknowledgmentMode())
+				.isEqualTo(ContainerProperties.ShareAcknowledgmentMode.IMPLICIT);
+	}
+
+	@Test
+	void shouldInvokeBasicMessageListener() throws Exception {
+		// Setup test data
+		ConsumerRecord<String, String> testRecord = new ConsumerRecord<>("test-topic", 0, 100L, "key", "value");
+		Map<TopicPartition, List<ConsumerRecord<String, String>>> records = new HashMap<>();
+		records.put(new TopicPartition("test-topic", 0), List.of(testRecord));
+		ConsumerRecords<String, String> testRecords = new ConsumerRecords<>(records, Map.of());
+
+		// Setup mocks
+		given(shareConsumerFactory.createShareConsumer(any(), any())).willReturn(shareConsumer);
+		given(shareConsumer.poll(any(Duration.class)))
+				.willReturn(testRecords)
+				.willReturn(ConsumerRecords.empty());
+
+		ContainerProperties containerProperties = new ContainerProperties("test-topic");
+		containerProperties.setMessageListener(messageListener);
+
+		ShareKafkaMessageListenerContainer<String, String> container =
+				new ShareKafkaMessageListenerContainer<>(shareConsumerFactory, containerProperties);
+
+		container.start();
+		Thread.sleep(1000);
```

Comment:
I'm not a fun of delays like this in tests.
Isn't there a way to assert on some thread barrier?
More over: do we still need this test class at all if you have all those comprehensive integration tests?
Isn't the functionality verified here covered over there as well?